doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+
+A prototype of the tasks UI, powered by the user's settings:
+
+![](tasks-suggestions.gif)
+
+(admittedly, the `TeachingTip` in that gif is a prototype and was later replaced
+with a better version.)
+
+In general, the Suggestions UI will present a list of elements to select from,
+near the text cursor. This control might be contain a text box for filtering
+these items (a "**palette**"), or it might not (a "**menu**").
+
+![An example of the menu mode](3121-suggestion-menu-2023-000.gif)
+
+#### Palette vs Menu
+
+Depending on how the suggestions UI is invoked, we may or may not want to
+display a text box for filtering these suggestions. Consider the Intellisense
+menu in Visual Studio. That's a UI that only allows for up/down for navigation
+(and enter/tab for selecting the suggestion).
+
+For suggestions driven by the Terminal, we'll display a filtering text box in
+the Suggestions UI. This is similar to the command palette's search - a fuzzy
+search to filter the contents. This is the "**palette**" style of the
+suggestions dialog.
+
+For completions driven by the shell, we should probably not display the
+filtering text box. This is the "**menu**" style of the suggestion dialog. The
+user is primarily interacting with the shell here, not the Terminal.
+
+> **Warning**
+> TODO! For discussion, possibly with a real UX designer.
+
+How should we handle completions here? Tab? Enter? Right-Arrow? Should we have
+an element selected when we open the menu, or should tab/enter only work once
+the user has used the arrows at least once? Sublime allows for <kbd>tab</kbd> to
+complete the suggestion immediately.
+
+Consider also that these suggestions might be provided by the shell, as the user
+is typing at a commandline shell. For something like PowerShell, the user might
+want to start typing a command and have it tab-complete based off the shell's
+tab expansion rules. PowerShell's inline suggestions use right-arrow to
+differentiate "use this suggestion" vs tab for "tab expand what I'm typing at
+the prompt". We should probably preserve this behavior.
+
+We probably don't want to provide different experiences for the **menu** version
+of the Suggestions UI vs. the **palette** version. In the palette version, the
+user won't be pressing tab to tab-complete at the shell - the focus is out of
+the of terminal and in the Suggestions UI. With the menu version, the focus is
+still "in the terminal", and users would expect tab to tab-complete.
+
+We will want to make sure that there's some semblance of consistency across our
+implementation for the Suggestions UI, our own Command Palette, VsCode's
+intellisense and their own implementation of shell-completions in the Terminal.
+
+> **Note** 
+> In my prototype, for the "Menu" mode, I accepted ALL of right-arrow, tab, and
+> enter as "accept completion", and any other key dismissed the UI. This _felt_
+> right for that mode. I'm not sure we could make the same call for "palette"

```

Comment:
These few paragraph argue against eachother. Suggestion: Make a call or lay them out in a closing paragraph as open options